doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+an element selected when we open the menu, or should tab/enter only work once
+the user has used the arrows at least once? Sublime allows for <kbd>tab</kbd> to
+complete the suggestion immediately.
+
+Consider also that these suggestions might be provided by the shell, as the user
+is typing at a commandline shell. For something like PowerShell, the user might
+want to start typing a command and have it tab-complete based off the shell's
+tab expansion rules. PowerShell's inline suggestions use right-arrow to
+differentiate "use this suggestion" vs tab for "tab expand what I'm typing at
+the prompt". We should probably preserve this behavior.
+
+We probably don't want to provide different experiences for the **menu** version
+of the Suggestions UI vs. the **palette** version. In the palette version, the
+user won't be pressing tab to tab-complete at the shell - the focus is out of
+the of terminal and in the Suggestions UI. With the menu version, the focus is
+still "in the terminal", and users would expect tab to tab-complete.
+
+We will want to make sure that there's some semblance of consistency across our
+implementation for the Suggestions UI, our own Command Palette, VsCode's
+intellisense and their own implementation of shell-completions in the Terminal.
+
+> **Note** 
+> In my prototype, for the "Menu" mode, I accepted ALL of right-arrow, tab, and
+> enter as "accept completion", and any other key dismissed the UI. This _felt_
+> right for that mode. I'm not sure we could make the same call for "palette"
+> mode, where we'd need tab for navigating focus. 
+
+### Implementation Details
+
+#### Fork the Command Palette
+
+We're largely going to start with the Command Palette to build the Suggestions
+UI[[1](#footnote-1)]. The Command Palette is already a control we've built for displaying a
+transient list of commands and dispatching them to the rest of the app.
+
+Currently, the Command Palette is a single static control, at the top-center of
+the Terminal window, and occupying a decent portion of the screen. For the
+Suggestions UI, we'll instead want to make sure that the control appears
+relative to the current cursor position.
+
+We'll start by taking the command palette, and copying it over to a new control.
+This will allow us to remove large chunks of code dealing with different modes
+(i.e. the tab switcher), and code dealing with prefix characters to switch
+modes.
+
+We'll need to make some small modifications to enable the Suggestions UI to
+* work as a text cursor-relative control
+* exist as a Flyout outside the bounds of the Terminal window
+* If the Suggestions UI is too close to the bottom of the screen, we'll need it to open
+  "upwards", with the search box at the _bottom_ and the list extending above it
+* prevent it from switching to command-line mode
+* display tooltips / `TeachingTip`s / some secondary flyout with a description
+  of the suggestion (if provided)

```

Comment:
I'd lean towards "fork", yeah. It's different enough that we won't want to pollute the O.G. command palette, and it may have to live in a different layer (depending on whether it's the App or the Control)